src/compiler/scala/tools/nsc/typechecker/Infer.scala

```diff
@@ -582,7 +582,7 @@ trait Infer extends Checkable {
         }
       }
       val targs = solvedTypes(tvars, tparams, varianceInTypes(formals), upper = false, lubDepth(formals) max lubDepth(argtpes))
-      if (settings.warnInferAny && context.reportErrors && !fn.isEmpty) {
+      if (settings.warnInferAny && !fn.isEmpty) {
```

Comment:
> I see that it silences the warning because it may want to rejigger the enclosing application. Letting it ignore that the context is non-erroring shows the warning.

Given that `reportErrors` is `!(bufferErrors || reporter.isThrowing)` my first thought is should this stay at least `!reporter.isThrowing`.

But more broadly, is it safe to ignore buffering?  `TryTwice` uses `inSilentMode` up to twice, but `InferMethodAlternativeTwice` (which extends `TryTwice`) adds to that by uses `inSilentMode` within that silent mode for each alternative in the overloaded type, and does so up to twice.  So that's potentially a lot of times we're running `isApplicable` - don't we risk false negatives?

Reply:
Yes, it is risky. I briefly tried "clearWarnings" up the stack, but that looked brittle to me.

It needs better infrastructure to track and then warn, but that won't happen here.

If a good example of false example turns up, that would help determine if this is too risky. Also see if the only downside is more warning suppression.

Reply:
My other thought was a warning is now issued that conceals a subsequent warning at the same pos.